covenant/covenant.go

```diff
@@ -43,58 +44,32 @@ type CovenantEmulator struct {
 
 	pk *btcec.PublicKey
 
-	cc clientcontroller.ClientController
-	kc *keyring.ChainKeyringController
+	signer Signer
+	cc     clientcontroller.ClientController
 
 	config *covcfg.Config
 	logger *zap.Logger
-
-	// input is used to pass passphrase to the keyring
-	input      *strings.Reader
-	passphrase string
 }
 
 func NewCovenantEmulator(
 	config *covcfg.Config,
 	cc clientcontroller.ClientController,
 	passphrase string,
```

Comment:
this is probably not needed as it is now part of the signer

Reply:
good catch